### ISSUE TITLE
CS1591 incorrectly reported for records defined using the short syntax

### ISSUE THREAD

**Issue by WilStead** [dotnet/roslyn#47144]
**Version Used**: Version 16.8.0 Preview 2.0

**Steps to Reproduce**:

1. Enable XML documentation
```xml
<PropertyGroup>
  <GenerateDocumentationFile>true</GenerateDocumentationFile>
</PropertyGroup>
```
2. (Taken directly from the [ASP.NET Core .NET 5 Preview 8 blog post](https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-5-preview-8/))
```c#
public record Person([Required] string Name, [Range(0, 150)] int Age);
```

**Expected Behavior**: No warnings

**Actual Behavior**:
**CS1591** reported for both parameters (properties?).

This might actually be a valid warning, except that there is no mechanism to add such documentation. For example:
```c#
public record Person
(
    /// <summary>
    /// The person's name.
    /// </summary>
    [Required] string Name
);
```
The above produces **CS1587** instead: XML comment is not placed on a valid language element.

I am also unaware of any tag which might be added to the record itself to document the parameters. For example:
```c#
/// <summary>
/// A person.
/// </summary>
/// <property cref="Name">
/// The person's name.
/// </property>
public record Person([Required] string Name);
````
This hypothetical syntax is invalid (produces **CS1574** and fails to remove **CS1591**), but I could imagine something like it being used to document the auto-generated properties in a way that satisfies **CS1591**.

**Comment by RikkiGibson** [comment]
This could require a language design decision: is a doc comment directly on the parameter valid, or is a new kind of doc comment on the record declaration required. FYI @jcouv @333fred

When I tried adopting records in a personal project it felt natural to me to put the doc comment directly on the parameter, since I was basically refactoring auto-property declarations into record parameters.

**Comment by jcouv** [comment]
xml doc work for records is tracked in https://github.com/dotnet/roslyn/issues/44571
That includes a summary of the plan (we'll allow `<param>` on the record type to carry over to the primary constructor parameters).

**Comment by 333fred** [comment]
Duplicate of #44571. I'll add a note to that issue to make sure we consider this warning, but we should have 1 issue tracking this.
